### ISSUE TITLE
Warnings will be hard errors from `clang-16`

### ISSUE THREAD

**Issue by RCasatta** [BlockstreamResearch/rust-secp256k1-zkp#85]
In the CI there are some warnings:

```
warning: secp256k1-zkp-sys@0.10.0: In file included from depend/secp256k1/src/secp256k1.c:23:
warning: secp256k1-zkp-sys@0.10.0: In file included from depend/secp256k1/src/assumptions.h:12:
warning: secp256k1-zkp-sys@0.10.0: depend/secp256k1/src/util.h:30:5: warning: implicitly declaring library function 'printf' with type 'int (const char *, ...)' [-Wimplicit-function-declaration]
warning: secp256k1-zkp-sys@0.10.0:     printf("{");
warning: secp256k1-zkp-sys@0.10.0:     ^
warning: secp256k1-zkp-sys@0.10.0: depend/secp256k1/src/util.h:30:5: note: include the header <stdio.h> or explicitly provide a declaration for 'printf'
```

If I understand correctly what's happening in a downstream library this will become hard errors from clang-16

Error in downstream library:

```
warning: In file included from depend/secp256k1/src/secp256k1.c:23:
warning: In file included from depend/secp256k1/src/assumptions.h:12:
warning: depend/secp256k1/src/util.h:30:5: error: call to undeclared library function 'printf' with type 'int (const char *, ...)'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
warning:     printf("{");
warning:     ^
warning: depend/secp256k1/src/util.h:30:5: note: include the header <stdio.h> or explicitly provide a declaration for 'printf'
warning: 1 error generated.

error: failed to run custom build command for `secp256k1-zkp-sys v0.10.0`

```

**Comment by apoelstra** [comment]
Should port https://github.com/rust-bitcoin/rust-secp256k1/pull/735 to here which deletes this (totally unused) code.

**Comment by RCasatta** [comment]
Fixed by #86
